test/E2ETests/SmokeTestsOnNanoServer.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using E2ETests.Common;
+using Microsoft.AspNetCore.Server.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace E2ETests
+{
+    public class SmokeTestsOnNanoServer : IDisposable
+    {
+        private readonly XunitLogger _logger;
+        private readonly RemoteDeploymentConfig _remoteDeploymentConfig;
+
+        public SmokeTestsOnNanoServer(ITestOutputHelper output)
+        {
+            _logger = new XunitLogger(output, LogLevel.Information);
+
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("remoteDeploymentConfig.json")
+                .AddEnvironmentVariables()
+                .Build();
+
+            _remoteDeploymentConfig = new RemoteDeploymentConfig();
+            configuration.GetSection("NanoServer").Bind(_remoteDeploymentConfig);
+        }
+
+        [ConditionalTheory, Trait("E2Etests", "NanoServer")]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [SkipIfEnvironmentVariableNotEnabled("RUN_TESTS_ON_NANO")]
+        [InlineData(ServerType.Kestrel, 5000)]
+        [InlineData(ServerType.WebListener, 5000)]
```

Comment:
> Aren't tests running in parallel? Could the ports conflict?

This assembly's tests are set to run tests in serial fashion.

> Also, could they conflict on the remote machine because the deployer kills all the processes with a certain name?
 
I am not clear about the question. The deployer would kill the `MusicStore` and any child processes of it and since these tests run serially, we should be good.